subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/oldresult/DefaultResolvedConfigurationBuilder.java

```diff
@@ -16,26 +16,25 @@
 
 package org.gradle.api.internal.artifacts.ivyservice.resolveengine.oldresult;
 
-import org.gradle.api.artifacts.Dependency;
 import org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphNode;
 
 import java.util.HashMap;
 import java.util.Map;
 
 public class DefaultResolvedConfigurationBuilder implements ResolvedConfigurationBuilder {
-    private final Map<Long, Dependency> modulesMap = new HashMap<Long, Dependency>();
+    private final Map<Long, Object> modulesMap = new HashMap<Long, Object>();
     private final TransientConfigurationResultsBuilder builder;
 
     public DefaultResolvedConfigurationBuilder(TransientConfigurationResultsBuilder builder) {
         this.builder = builder;
     }
 
     @Override
-    public void addFirstLevelDependency(Dependency moduleDependency, DependencyGraphNode dependency) {
+    public void addFirstLevelDependency(Object moduleDependencyOrConstraint, DependencyGraphNode dependency) {
```

Comment:
This raises an interesting question: if A dependsOn B dependsOn C, and A has a constraint on C, does that make C a "first level" dependency of A? From Adam's comment I'd say not.